### ISSUE TITLE
Some questions on the training code

### ISSUE THREAD

**Issue by stalin18** [PolyLiYJ/StyleGuard#3]
Hi! I have some questions in the training code, I will be glad if you could clarify those.

1. In Algorithm 1 line 6, it is mentioned that copyed UNet model weights are updated using X_p (protected image); however, in L885 in `stylegaurd.py` `clean_data_batch` (clean) is used. Which one is correct, the paper or the code?

2. About the algorithm, in L878 `f` is deep copied into `f_sur`. Then `f_sur` is trained on clean images, then using `f_sur` `pgd_attack` is performed to obtained `perturbed_data_f_batch`. Now `f` is trained on `perturbed_data_f_batch` and its weights are saved. In the next iteration, `f_sur` gets loaded with these new saved weights of `f` and then trained on clean images. As I understand a strong perturbation needs to be learned which is effective to prevent DreamBooth training, so performing `pgd_attack` against `f` trained on perturbed image should be sufficient. Why perturbed image trained `f` is loaded into `f_sur` and trained on clean images again? This makes `pgd_attack` simple to learn, right? Please help to understand.

3. The `pgd_attack` is learned separately for DreamBooth denoising loss (L888) and upscalar loss (L923) in different loops. Why not compute and denoising and upscale loss together and add then as mentioned in paper in Eq. 7 and train? Could you please clarify?

**Comment by PolyLiYJ** [comment]
Hi,

Thanks for your question,

Our experimental results indicate minimal performance differences between using clean versus perturbed data for fine-tuning the copied UNet model. Quantitative evaluation shows FID scores of 365.68 (clean data) and 356.42 (perturbed data). 

In the original Anti-Dreambooth paper's methodology, which employed clean data for UNet fine-tuning. To maintain methodological consistency within our paper, we have modified the implementation to use perturbed data throughout our framework.

Additionally, we implemented the denoising and upscaler loss calculations into separate processing loops to reduce GPU memory cost.

Best Regards

Yanjie
